weaver/main.go

```diff
@@ -163,7 +183,45 @@ func initiateConnections(router *weave.Router, peers []string) {
 	}
 }
 
-func handleHTTP(router *weave.Router, httpAddr string) {
+func createAllocator(router *weave.Router, apiPath string, iprangeCIDR string, quorum uint) *ipam.Allocator {
+	allocator, err := ipam.NewAllocator(router.Ourself.Peer.Name, router.Ourself.Peer.UID, iprangeCIDR, quorum)
+	if err != nil {
+		log.Fatal(err)
+	}
+	allocator.SetInterfaces(router.NewGossip("IPallocation", allocator))
```

Comment:
this is racy. The router has already been started, so you could receive IPAM gossip from that point onwards, which will result in errors in the logs. I think.

Really the allocator and router should be wired up before calling `Start()` on either of them.